src/WebJobs.Script.WebHost/Helpers/TableStorageHelpers.cs

```diff
@@ -74,38 +73,49 @@ internal static void QueueBackgroundTablePurge(CloudTable currentTable, CloudTab
             });
         }
 
-        internal static async Task DeleteOldTablesAsync(CloudTable currentTable, CloudTableClient tableClient, string tableNamePrefix, ILogger logger)
+        internal static async Task DeleteOldTablesAsync(TableClient currentTable, TableServiceClient tableClient, string tableNamePrefix, ILogger logger)
         {
             var tablesToDelete = await ListOldTablesAsync(currentTable, tableClient, tableNamePrefix);
             logger.LogDebug($"Deleting {tablesToDelete.Count()} old tables.");
             foreach (var table in tablesToDelete)
             {
                 logger.LogDebug($"Deleting table '{table.Name}'");
-                await table.DeleteIfExistsAsync();
+                await tableClient.DeleteTableAsync(table.Name);
                 logger.LogDebug($"{table.Name} deleted.");
             }
         }
 
-        internal static async Task<IEnumerable<CloudTable>> ListOldTablesAsync(CloudTable currentTable, CloudTableClient tableClient, string tableNamePrefix)
+        internal static async Task<IEnumerable<TableClient>> ListOldTablesAsync(TableClient currentTable, TableServiceClient tableClient, string tableNamePrefix)
         {
             var tables = await ListTablesAsync(tableClient, tableNamePrefix);
             return tables.Where(p => !string.Equals(currentTable.Name, p.Name, StringComparison.OrdinalIgnoreCase));
         }
 
-        internal static async Task<IEnumerable<CloudTable>> ListTablesAsync(CloudTableClient tableClient, string tableNamePrefix)
+        internal static async Task<IEnumerable<TableClient>> ListTablesAsync(TableServiceClient tableClient, string tableNamePrefix)
         {
-            List<CloudTable> tables = new List<CloudTable>();
-            TableContinuationToken continuationToken = null;
+            // Azure.Data.Tables doesn't have a direct way to list tables with a prefix
+            AsyncPageable<TableItem> tablesQuery = tableClient.QueryAsync(p => p.Name.CompareTo(tableNamePrefix) >= 0);
+            var tables = new List<TableClient>();
 
-            do
+            await foreach (var page in tablesQuery.AsPages())
             {
-                var results = await tableClient.ListTablesSegmentedAsync(tableNamePrefix, continuationToken);
-                continuationToken = results.ContinuationToken;
-                tables.AddRange(results.Results);
+                foreach (var table in page.Values)
+                {
+                    tables.Add(tableClient.GetTableClient(table.Name));
+                }
             }
-            while (continuationToken != null);
 
             return tables;
         }
+
+        internal static async Task<bool> CheckIfTableExistsAsync(TableServiceClient tableClient, string tableName)
```

Comment:
This isn't used, but also checking for existence then operating on the table could be a race condition: what if the table is deleted between the two statements?

Reply:
I agree, on those situations we would have a conflict/error when using it. I added it to migrate something that we do in a test but then didn't go this way and forgot to remove it.